start.py

```diff
@@ -558,40 +563,54 @@ def _command_condition(deployment, command_id, command_description):
     return not active and success
 
 
-# TODO: Get rid of the excess amount of code duplication between the following
-# function and _configure_kudu.
-def _configure_kafka(deployment, cluster, args):
+def _deactivate_service_parcel(deployment, cluster, product, version):
+    parcel_found_state = 'not_found'
     for parcel in deployment.cluster(DEFAULT_CLUSTER_NAME).parcels:
-        if parcel.product == 'KAFKA' and parcel.stage in 'ACTIVATED':
-            parcel_kafka_version = parcel.version.split('-')[0]
-            if parcel_kafka_version == args.kafka_version:
-                logger.info('Detected Kafka version matches specified version. Continuing ...')
+        if parcel.product == product and parcel.stage in 'ACTIVATED':
+            parcel_version = parcel.version.split('-')[0]
+            if parcel_version == version:
+                logger.info('Detected parcel version matches specified version. Continuing ...')
+                parcel_found_state = 'matched_active_version'
             else:
                 parcel.deactivate()
+                parcel_found_state = 'is_deactivated'
+            # Only one parcel can be activated at a time, so once one is found, our work is done.
+            break
+    return parcel_found_state
 
-                for config in deployment.get_cm_config():
-                    if config['name'] == 'REMOTE_PARCEL_REPO_URLS':
-                        break
-                else:
-                    raise Exception('Failed to find remote parcel repo URLs configuration.')
-                parcel_repo_urls = config['value']
 
-                kafka_parcel_repo_url = '{}/{}'.format(KAFKA_PARCEL_REPO_URL, args.kafka_version)
-                logger.debug('Adding Kafka parcel repo URL (%s) ...', kafka_parcel_repo_url)
-                deployment.update_cm_config(
-                    {'REMOTE_PARCEL_REPO_URLS': '{},{}'.format(parcel_repo_urls,
-                                                               kafka_parcel_repo_url)}
-                )
+def _install_parcel_from_remote_repo(deployment, product, version):
+    for config in deployment.get_cm_config():
+        if config['name'] == 'REMOTE_PARCEL_REPO_URLS':
+            break
+    else:
+        raise Exception('Failed to find remote parcel repo URLs configuration.')
+    parcel_repo_urls = config['value']
+
+    service_parcel_repo_url = '{}/{}'.format(SERVICE_PARCEL_REPO_URLS[product],
+                                             KUDU_PARCEL_VERSIONS[version] if product == 'KUDU' else version)
```

Comment:
Building this Kudu check into the function seems a bit hacky. Could we figure out a way to have it handle this lookup in the calling function instead of here?

Reply:
Done.